src/main/java/uk/nhs/hee/tis/trainee/sync/mapper/AggregateMapper.java

```diff
@@ -0,0 +1,145 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.trainee.sync.mapper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingConstants.ComponentModel;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.ReportingPolicy;
+import uk.nhs.hee.tis.trainee.sync.dto.AggregateCurriculumMembershipDto;
+import uk.nhs.hee.tis.trainee.sync.dto.AggregateProgrammeMembershipDto;
+import uk.nhs.hee.tis.trainee.sync.model.Curriculum;
+import uk.nhs.hee.tis.trainee.sync.model.CurriculumMembership;
+import uk.nhs.hee.tis.trainee.sync.model.Programme;
+import uk.nhs.hee.tis.trainee.sync.model.ProgrammeMembership;
+import uk.nhs.hee.tis.trainee.sync.model.Record;
+
+/**
+ * A mapper for creating DTO aggregating from multiple data types.
+ */
+@Mapper(componentModel = ComponentModel.SPRING, unmappedTargetPolicy = ReportingPolicy.IGNORE)
+public interface AggregateMapper {
+
+  /**
+   * Create an aggregate curriculum DTO.
+   *
+   * @param curriculum           The curriculum to aggregate from.
+   * @param curriculumMembership The curriculum membership to aggregate from.
+   * @return The aggregated curriculum DTO.
+   */
+  @Mapping(target = "curriculumTisId", source = "curriculum.tisId")
+  @Mapping(target = "curriculumName", source = "curriculum.data.name")
+  @Mapping(target = "curriculumSubType", source = "curriculum.data.curriculumSubType")
+  @Mapping(target = "curriculumMembershipId", source = "curriculumMembership.tisId")
+  @Mapping(target = "curriculumStartDate", source = "curriculumMembership.data.curriculumStartDate")
+  @Mapping(target = "curriculumEndDate", source = "curriculumMembership.data.curriculumEndDate")
+  AggregateCurriculumMembershipDto toAggregateCurriculumMembershipDto(Curriculum curriculum,
+      CurriculumMembership curriculumMembership);
+
+  /**
+   * Create an aggregate programme membership DTO.
+   *
+   * @param programmeMembership The programme membership to aggregate from.
+   * @param programme           The programme to aggregate from.
+   * @param curricula           The curricula to aggregate from.
+   * @return The aggregated programme membership DTO.
+   */
+  @Mapping(target = "tisId", source = "programmeMembership.uuid")
+  @Mapping(target = "personId", source = "programmeMembership.personId")
+  @Mapping(target = "programmeMembershipType",
+      source = "programmeMembership.programmeMembershipType")
```

Comment:
Is it obvious why some of these mappings use *.data.* and others don't? I'm not quite clear on the rationale, e.g. `curriculumMembership.data.curriculumStartDate` above, but `programmeMembership.programmeStartDate` below

Reply:
`ProgrammeMembership` was converted to a proper POJO in #351, so doesn't have the data map anymore.